plugins/content_distribution/providers/cross_kaltura/lib/batch/CrossKalturaDistributionEngine.php

```diff
@@ -142,12 +166,27 @@ protected function initClients(KalturaCrossKalturaDistributionProfile $distribut
 
 		// init target client
 		$targetClientConfig = new KalturaConfiguration($distributionProfile->targetAccountId);
-		$targetClientConfig->serviceUrl = $distributionProfile->targetServiceUrl;
+		$targetClientConfig->serviceUrl = $this->getTargetServiceUrl($distributionProfile->targetServiceUrl);
 		$targetClientConfig->setLogger($this);
 		$this->targetClient = new KalturaClient($targetClientConfig);
 		$ks = $this->targetClient->user->loginByLoginId($distributionProfile->targetLoginId, $distributionProfile->targetLoginPassword, $distributionProfile->targetAccountId, 86400, 'disableentitlement');
 		$this->targetClient->setKs($ks);
 	}
+	
+	private function getTargetServiceUrl($serviceUrl)
+	{
+		if(!$this->targetServiceUrlRegex)
+		{
+			return $serviceUrl;
+		}
+		
+		if (!preg_match($this->targetServiceUrlRegex, $serviceUrl, $matches))
```

Comment:
need to remove the $matches from here too :)

Reply:
Done.